From ef8e6beaebcfc3e143b249ba76c177fbabd3b4ea Mon Sep 17 00:00:00 2001
From: Yichen <yichenz@chromium.org>
Date: Thu, 23 May 2024 23:44:41 +0000
Subject: [PATCH] ui: fix gcc/libstdc++ build break

Bug: 340815321
Change-Id: Ibf443f9f16dfff164e1f4d49a58582fb7a6aca1d
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5565320
Reviewed-by: Xiyuan Xia <xiyuan@chromium.org>
Commit-Queue: Yichen Zhou <yichenz@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1305426}
---
 ui/color/color_provider.cc | 3 ++-
 ui/color/color_provider.h  | 3 +--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/ui/color/color_provider.cc b/ui/color/color_provider.cc
index 1597715de1f63c..3e0ffed9f4d4c4 100644
--- a/ui/color/color_provider.cc
+++ b/ui/color/color_provider.cc
@@ -118,7 +118,8 @@ class ColorProvider::ColorProviderInternal {
 ////////////////////////////////////////////////////////////////////////////////
 // ColorProvider:
 
-ColorProvider::ColorProvider() = default;
+ColorProvider::ColorProvider()
+    : internal_(std::make_unique<ColorProviderInternal>()) {}
 
 ColorProvider::ColorProvider(ColorProvider&&) = default;
 
diff --git a/ui/color/color_provider.h b/ui/color/color_provider.h
index 50fff33ae6ab6d..6ac6ed6938cf24 100644
--- a/ui/color/color_provider.h
+++ b/ui/color/color_provider.h
@@ -65,8 +65,7 @@ class COMPONENT_EXPORT(COLOR) ColorProvider {
   // bind to it.
   class ColorProviderInternal;
 
-  std::unique_ptr<ColorProviderInternal> internal_ =
-      std::make_unique<ColorProviderInternal>();
+  std::unique_ptr<ColorProviderInternal> internal_;
 };
 
 }  // namespace ui
